modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/object/dynamic/data/mapping/form/field/type/internal/attachment/AttachmentDDMFormFieldTemplateContextContributor.java

```diff
@@ -85,12 +91,72 @@ public Map<String, Object> getParameters(
 				})
 		).put(
 			"url", _getURL(ddmFormField, ddmFormFieldRenderingContext)
-		).putAll(
+		).build();
+
+		if (FeatureFlagManagerUtil.isEnabled("LPD-32050")) {
+			DDMForm ddmForm = ddmFormField.getDDMForm();
+			boolean localizedObjectField = GetterUtil.getBoolean(
+				ddmFormField.getProperty("localizedObjectField"));
+
+			return HashMapBuilder.<String, Object>put(
+				"fileEntryProperties",
+				() -> {
+					if (localizedObjectField) {
+						JSONObject localizedValueJSONObject =
+							DDMFormFieldValueUtil.getValueJSONObject(
+								ddmFormFieldRenderingContext);
+
+						Map<String, Object> localizedValue =
+							localizedValueJSONObject.toMap();
+
+						for (Map.Entry<String, Object> entry :
+								localizedValue.entrySet()) {
+
+							localizedValue.put(
+								entry.getKey(),
+								_getFileEntryProperties(
+									ddmFormField,
+									ddmFormFieldRenderingContext.
+										getHttpServletRequest(),
+									GetterUtil.getLong(entry.getValue())));
+						}
+
+						return _jsonFactory.createJSONObject(localizedValue);
+					}
+
+					return _getFileEntryProperties(
+						ddmFormField,
+						ddmFormFieldRenderingContext.getHttpServletRequest(),
+						GetterUtil.getLong(
+							ddmFormFieldRenderingContext.getValue()));
+				}
+			).put(
+				"localizedObjectField", localizedObjectField
+			).put(
+				"value",
+				() -> {
+					if (localizedObjectField) {
+						return DDMFormFieldValueUtil.getValueJSONObject(
+							ddmFormFieldRenderingContext);
+					}
+
+					return ddmFormFieldRenderingContext.getValue();
+				}
+			).putAll(
+				DDMFormFieldTemplateContextContributorUtil.getLocaleMap(
+					ddmForm.getDefaultLocale())
+			).putAll(
+				parameters
+			).build();
+		}
+
+		parameters.putAll(
 			_getFileEntryProperties(
 				ddmFormField,
 				ddmFormFieldRenderingContext.getHttpServletRequest(),
-				GetterUtil.getLong(ddmFormFieldRenderingContext.getValue()))
-		).build();
+				GetterUtil.getLong(ddmFormFieldRenderingContext.getValue())));
```

Comment:
Will we maintain this after removing the FF? I would like to know if we will need both in the future.

Reply:
If yes, it is strange because it will have duplicated data in `parameters`.

Reply:
I haven't approved or rejected this yet because this logic seems unusual. Should this code be inside the else block instead of always executing?

Reply:
I updated it to use an if-else statement. Thank you!